### PR TITLE
feat: Integration with Dremio

### PR DESCRIPTION
**Pull request by joshthoward**
### Summary of Changes

Added a couple of things around the same theme of Dremio integration... 

1. Added an icon for Dremio datasets
2. Added link to Dremio tables as an optional data source
3. Added example preview querying for querying Dremio using Arrow Flight

### Tests

N/A; current tests are satisfied and preview examples don't have tests. 

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [-] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [-] PR includes a summary of changes, including screenshots of any UI changes.
- [-] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [-] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [-] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/components/AvatarLabel/index.tsx [hunk 1]
```diff
@@ -11,24 +11,33 @@ export interface AvatarLabelProps {
...
     <span className={`avatar-label body-2 ${labelClass}`}>{label}</span>
   </div>
 );
 
 AvatarLabel.defaultProps = {
   labelClass: 'text-secondary',
   label: '',
+  round: true,
   src: '',
 };
```
Comment (Golodhros) [inline amundsen_application/static/js/components/AvatarLabel/index.tsx:42]: Would you mind to move this into the destructuring in lines 19-23?
We are going to deprecate using `.defaultProps` altogether, there isn't a good reason for them anymore.

**Review by Golodhros — COMMENTED** [review]
JS looks good, only a minor request.

File: amundsen_application/static/js/components/AvatarLabel/index.tsx (hunk 1 above)
Comment (joshthoward) [reply amundsen_application/static/js/components/AvatarLabel/index.tsx:42]: @Golodhros Fair enough. I've got this fixed in my local copy. Before pushing, can you help me work around this `eslint` error? 

https://github.com/amundsen-io/amundsenfrontendlibrary/pull/859/checks?check_run_id=1647122266#step:5:19

Comment (Golodhros) [reply amundsen_application/static/js/components/AvatarLabel/index.tsx:42]: Oh, that seems like a false positive we should fix. For now, just run `npm run betterer:update` to get the exception approved. 
I will try to look into it this week.

Thanks!

**Review by Golodhros — APPROVED** [review]
LGTM

Thanks!

**Review by allisonsuarez — APPROVED** [review]
LGTM!

**Review by feng-tao — APPROVED** [review]

**Comment by feng-tao** [comment]
thanks!

**Pull request merged**